javagen/src/main/java/com/azure/autorest/template/StreamSerializationModelTemplate.java

```diff
@@ -263,13 +294,41 @@ private static void serializeJsonProperty(JavaBlock methodBlock, ClientModelProp
             return;
         }
 
+        if (isJsonMergePatch) {
+            methodBlock.ifBlock(String.format("%s !=null", property.getName()), codeBlock -> {
+                serializeJsonProperty(codeBlock, property, serializedName,
+                        false);
+            }).elseIfBlock(String.format("updatedProperties.contains(\"%s\")", property.getName()), codeBlock -> {
+                codeBlock.line(String.format("jsonWriter.writeNullField(\"%s\");", property.getSerializedName()));
+            });
+        } else {
+            serializeJsonProperty(methodBlock, property, serializedName, fromSuperType);
+        }
+    }
+
+    /**
+     * Serializes a non-flattened, non-additional properties JSON property.
+     * <p>
+     * If the JSON property needs to be flattened or is additional properties this is a no-op as those require special
+     * handling that will occur later.
+     *
+     * @param methodBlock    The method handling serialization.
+     * @param property       The property being serialized.
+     * @param serializedName The serialized JSON property name. Generally, this is just the {@code property property's}
+     *                       serialized name but if a flattened property is being serialized it'll be the last segment of the flattened JSON
+     *                       name.
+     * @param fromSuperType  Whether the property is defined by a super type of the model. If the property is declared by
+     *                       a super type a getter method will be used to retrieve the value instead of accessing the field directly.
+     */
+    private static void serializeJsonProperty(JavaBlock methodBlock, ClientModelProperty property,
+                                              String serializedName, boolean fromSuperType) {
         IType clientType = property.getClientType();
         IType wireType = property.getWireType();
         String propertyValueGetter;
         if (fromSuperType) {
             propertyValueGetter = (clientType != wireType)
-                ? wireType.convertFromClientType(property.getGetterName() + "()")
-                : property.getGetterName() + "()";
+                    ? wireType.convertFromClientType(property.getGetterName() + "()")
+                    : property.getGetterName() + "()";
```

Comment:
avoid this kind of unnecessary formatting change

Reply:
I will revert back those spaces. 
BTW, Do we have command to do code format, like ts have `npm run format`, so there will not have unnecessary formatting change.

Reply:
No. checkstyle is all we need for now.

Just give other dev the freedom to write their own code, as long as it does not against checkstyle.